internal/rhsm/secrets.go

```diff
@@ -0,0 +1,108 @@
+package rhsm
+
+import (
+	"fmt"
+	"io/ioutil"
+	"strings"
+
+	"gopkg.in/ini.v1"
+)
+
+type subscription struct {
+	id            string
+	baseurl       string
+	sslCACert     string
+	sslClientKey  string
+	sslClientCert string
+}
+
+// Subscriptions encapsulates all available subscriptions from the
+// host system.
+type Subscriptions struct {
+	available []subscription
+}
+
+// RHSMSecrets represents a set of CA certificate, client key, and
+// client certificate for a specific repository.
+type RHSMSecrets struct {
+	SSLCACert     string
+	SSLClientKey  string
+	SSLClientCert string
+}
+
+// LoadSystemSubscriptions loads all the available subscriptions.
+func LoadSystemSubscriptions() (*Subscriptions, error) {
+	availableSubscriptionsFile := "/etc/yum.repos.d/redhat.repo"
```

Comment:
Is the special semantics of this file documented anywhere we could refer to? Might make sense to explain in some detail. Could people use other repo files for the same purpose, or is this special?

Reply:
It is not different from any other repo file. I wrote this fact into a comment.

Reply:
So the reason that it is special is that it is the file written by RHSM, right? I.e., it does not make sense for us to support the same logic for other repo files, as we are explicitly targeting RHSM?